ui/components/GHImport.js

```diff
@@ -0,0 +1,155 @@
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import { URLValidator } from '../utils/URLValidator';
+
+const getMuiTheme = () => createTheme({
+  palette : {
+    primary : {
+      main : "#607d8b"
+    }
+  },
+  overrides : {
+    MuiGrid : {
+      input : {
+        color : '#607d8b'
+      }
+    },
+  }
+})
+
+
+const styles = makeStyles((theme) => ({
+  paper : {
+    position : 'absolute',
+    width : 600,
+    backgroundColor : theme.palette.background.paper,
+    border : '0px solid #000',
+    boxShadow : theme.shadows[5],
+    padding : theme.spacing(2, 4, 3),
+    top : '50%',
+    left : '50%',
+    transform : `translate(-50%, -50%)`,
+    borderRadius : 10,
+  },
+  grid : {
+    width : '100%'
+  },
+  GridTitle : {
+    textAlign : 'center'
+  },
+  formControl : {
+    margin : theme.spacing(1),
+    minWidth : 120,
+  },
+  ghGroups : {
+    padding : theme.spacing(0, 13, 0)
+  },
+  optionImport : {
+    textAlign : 'center'
+  }
+
+}));
+import GenericModal from "./GenericModal";
+
+
+const GHImport = ({ onSubmit }) => {
+  const classes = styles();
+  const [open, setOpen] = React.useState(false);
+  const [input, setInput] = React.useState('');
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+  const validURL = (str) => {
+    return URLValidator(str);
+  }
+  const handleError = (input) => {
+    console.log(input + ' is not valid a valid repository')
+  }
+  const handleSubmit = () => {
+    validURL(input) ? onSubmit(input) : handleError(input);
+    handleClose()
+  }
+
+  return (
+    <React.Fragment>
+      <label htmlFor="github-import-button">
+
+        <Tooltip title="Github Import">
+          <IconButton aria-label="Github-Import" component="span" onClick={handleOpen}>
+            <GitHubIcon />
+          </IconButton>
+        </Tooltip>
+        <GenericModal
+          open={open}
+          handleClose={handleClose}
+          Content={
+            <MuiThemeProvider theme={getMuiTheme()}>
+              <div
+                className={classes.paper}
+              >
+                <Grid
+                  container spacing={2}
+                >
+                  <Grid
+                    className={classes.GridTitle}
+                    item xs={12}>
+                    <h2 id="simple-modal-title">Import using Github</h2>
+                  </Grid>
+                  <div className={classes.ghGroups}>
+                    <FormControl className={classes.formControl}>
+                      <InputLabel htmlFor="grouped-native-select">Repository</InputLabel>
+                      <Select defaultValue="" id="grouped-select">
+                        <MenuItem value="">
+                          <em>None</em>
+                        </MenuItem>
+                        <MenuItem value={1}>service-mesh-patterns</MenuItem>
```

Comment:
@utkarsh-pro Is there a way I can directly target the repository and the branch(on select), or it should be done with the function handler for GitHub import?

Reply:
+ what all other things for now I can add as per the UX in the UI ?

Reply:
Why are we using dropdown here? Are there limited repos that we can import the files from?
From what I understood from the call, we need to take owner, repo and branch as inputs, dir path and one check box as inputs. So like 4 input fields and one check box, right?

Reply:
We have currently two repos that we will be using to import the patterns and filters provided by Meshery for now.
since the repos which Meshery provides can remain in the dropdown, and to take the owner repo we may use an input URL option.  yes we could replace the branch dropdown selection with the checkbox.

Reply:
I'm sorry I didn't get why we need to replace branch dropdown with check box. We just need to use `TextField` instead of dropdown right? 

Owner: TextField (just take the owner github id) 
Repo: Dropdown 
Branch: TextField 
Dir Path: TextField 
Recurrsive? : Checkbox
This will work, right?

Reply:
yes, you are right @Meghana-12 , but I think the branch will always, remain master or main only, yes it may change when it's users' own repository(preference). I did the previous one based on repos branches provided by meshery. I'll update up in the UI

Reply:
@Chadha93 a user can import patterns from any repository and any branch and I think we should give them a text box rather than a drop down.

Reply:
Gotcha!